src/client.ts

```diff
@@ -495,13 +497,15 @@ export function createClient(options: ClientOptions): Client {
             // either the canceller will be called and the promise resolved
             // or the socket closed and the promise rejected
             await throwOnCloseOrWaitForCancel(() => {
-              // send complete message to server on cancel
-              socket.send(
-                stringifyMessage<MessageType.Complete>({
-                  id: id,
-                  type: MessageType.Complete,
-                }),
-              );
+              if (!completed && socket.readyState === WebSocketImpl.OPEN) {
```

Comment:
Why check for the open state?

The cleanup callback is called exclusively if the user cancels the subscription. You can cancel only an active connection. If the socket closes at any point in time, the `throwOnCloseOrWaitForCancel` will throw and the cleanup will never be called.

```suggestion
              if (!completed) {
```

Reply:
I added this check for the socket state because I noticed some errors in my console where socket state was already CLOSING or CLOSED at this point.

Reply:
Resolved in favour of: https://github.com/enisdenjo/graphql-ws/pull/65#discussion_r522140655.

Reply:
Okay then, this has be handled! The socket _should_ still be open here. Or maybe my reasoning is wrong. I'll come back to this later...

Reply:
Do you want me to remove the check?

Reply:
I am currently not fully concentrated on this TBH, am swamped by my day job.

But, I will come back to this later investing a proper thought. I hope you understand.

Reply:
I fully understand and I apologize for flooding you with message. I'm focused on this exactly because it is overlapping with my day job ATM.  

I'll remove the check as part of this PR  and I'll try debugging it later to see what is happening.

Reply:
No worries! I respond when I can. I have unresolved this discussion, because I'd like to get to the bottom of this first.

As mentioned in the https://github.com/enisdenjo/graphql-ws/pull/65#issuecomment-726103670, the client should not complain about the server not sending the complete message. So, you can leave your server implementation be and the client will still work.

**However,** if this is not the case for you - I will be happy to merge just the client fix to elevate your block! In this case, please put the readyState check back. Also, if we merge, I would kindly ask you to open an issue for the server fix too (you can leave the implementation to me).

Reply:
The client doesn't have any problems with the server not sending the Complete message.

I'm making the enhancements to the server side and unit tests. I'll push an updated patch soon.

Reply:
The issue with socket state seems to be unrelated to these changes, so I am planning on investigating that separately.

Reply:
Just a short question: you were using the server from this lib (not your own) when the state issues were occurring, right?

Reply:
No. The state warning is happening with our own server, so it could be an issue with our server.  I'm debugging it now to see what is happening.  If it turns out to be an issue on the client I'll open a separate issue.

Reply:
But, is it happening on this server? If not, I would say the issue is not related. The only difference in the client with this PR is that it has -1 send, I guess these CLOSED and CLOSING warnings would have been noticed before - or caught by tests.

Still, if it indeed turns out to be a client issue, please open an additional issue then.

Reply:
I found the issue and opened https://github.com/enisdenjo/graphql-ws/issues/66